bluesky/callbacks/core.py

```diff
@@ -50,6 +78,125 @@ def stop(self, doc):
         pass
 
 
+class RunRouter(CallbackBase):
+    """
+    Routes documents, by run, to callbacks it creates from factory functions.
+
+    A RunRouter is callable, and it is has the signature ``router(name, doc)``,
+    suitable for subscribing to the RunEngine.
+
+    The RunRouter maintains a list of factory functions with the signature
+    ``callback_factory(start_doc)``. When the router receives a RunStart
+    document, it passes it to each ``callback_factory`` function. Each factory
+    should return ``None`` ("I am not interested in this run,") or a callback
+    function with the signature ``cb(name, doc)``. All future documents related
+    to that run will be forwarded to ``cb``. When the run is complete, the
+    RunRouter will drop all its references to ``cb``. It is up to
+    ``callback_factory`` whether to return a new callable each time (having a
+    lifecycle of one run, garbage collected thereafter), or to return the same
+    object every time, or some other pattern.
+    
+    To summarize, the RunRouter's promise is that it will call each
+    ``callback_factory`` with each new RunStart document and that it will
+    forward all other documents from that run to whatever ``callback_factory``
+    returns (if not None).
+
+    Parameters
+    ----------
+    callback_factories : list
+        A list of callables with the signature:
+
+            callback_factory(start_doc)
+
+        which should return ``None`` or another callable with this signature:
+
+            callback(name, doc)
+    """
+    def __init__(self, callback_factories):
```

Comment:
would it be possible to accept `**kwargs` to be passed to the callback factories?